uvicorn/main.py

```diff
@@ -549,6 +549,9 @@ async def on_tick(self, counter) -> bool:
         return False
 
     async def shutdown(self, sockets=None):
+        # Send the lifespan shutdown before event, allowing looping requests to exit
+        await self.lifespan.shutdown_notice()
```

Comment:
Isn't it race-y to close long-running requests here, before the server stopped receiving new ones? That still leaves a time window (however brief) where a new connection could be opened, thus preventing the server from closing again. I know the proposed ASGI lifespan RFC says that's when the signal is meant to be emitted, but is there a reason for that?

Reply:
That's a great point! I adjusted the implementation and the RFC language to reflect moving when the event is triggered.